capturable/src/iosMain/kotlin/dev/shreyaspatil/capturable/Capturable.ios.kt

```diff
@@ -0,0 +1,114 @@
+
+    override fun onAttach() {
+        super.onAttach()
+        coroutineScope.launch {
+            observeCaptureRequestsAndServe()
+        }
+    }
+
+    actual fun updateController(controller: CaptureController) {
+        currentController.value = controller
+    }
+
+    @OptIn(ExperimentalCoroutinesApi::class)
+    private suspend fun observeCaptureRequestsAndServe() {
+        currentController
+            .flatMapLatest { it.captureRequests }
+            .collect { request ->
+                val completable = request.imageBitmapDeferred
+                try {
+                    val picture = getCurrentContentAsPicture()
+                    val bitmap = withContext(Dispatchers.Default) {
+                        picture.asBitmap(request.config)
+                    }
+
+                    completable.complete(bitmap.asComposeImageBitmap())
+                } catch (error: Throwable) {
+                    completable.completeExceptionally(error)
+                }
+            }
+    }
+
+    private suspend fun getCurrentContentAsPicture(): Picture {
+        return drawCanvasIntoPicture()
+    }
+
+    private suspend fun drawCanvasIntoPicture(): Picture {
+        val pictureDrawn = CompletableDeferred<Unit>()
+        val pictureRecorder = PictureRecorder()
+        var picture = Picture.makePlaceholder(org.jetbrains.skia.Rect(0f, 0f, 0f, 0f))
+
+        val delegatedNode = delegate(
+            CacheDrawModifierNode {
+                val width = this.size.width
+                val height = this.size.height
+
+                onDrawWithContent {
+                    val pictureCanvas = pictureRecorder.beginRecording(org.jetbrains.skia.Rect.makeWH(width, height))
+
+                    draw(this, this.layoutDirection, pictureCanvas.asComposeCanvas(), this.size) {
+                        this@onDrawWithContent.drawContent()
+                    }
+
+                    picture = pictureRecorder.finishRecordingAsPicture()
+
+                    drawIntoCanvas { canvas ->
+                        (canvas.nativeCanvas as org.jetbrains.skia.Canvas).drawPicture(picture)
+                        // Notify that picture is drawn
+                        pictureDrawn.complete(Unit)
+                    }
+                }
+            }
+        )
+        pictureDrawn.await()
+
+        undelegate(delegatedNode)
+
+        return picture;
+    }
+}
+
+
+private fun Picture.asBitmap(config: ColorType = ColorType.ARGB_4444): Bitmap {
+    val bitmap = Bitmap()
+    val ci = ColorInfo(config, ColorAlphaType.OPAQUE, ColorSpace.sRGB)
+    val width = this@asBitmap.cullRect.width
+    val height = this@asBitmap.cullRect.height
+    bitmap.setImageInfo(ImageInfo(ci, width.toInt(), height.toInt()))
+    bitmap.allocN32Pixels(width.toInt(), height.toInt())
+    val canvas = org.jetbrains.skia.Canvas(bitmap)
+    canvas.drawPicture(this)
+    bitmap.setImmutable()
+    return bitmap
+}
```

Comment:
Also, instead of creating duplicate `Capturable` implementation across multiple platforms, can't we just create a platform-specific implementation for `CacheDrawModifierNode`?

Reply:
I was leaving room for more stronger changes that might come. But I think we can simplify it for now if you think that'd be better.
I'm pro the current setup so we don't have to do alot of expect/actuals

Reply:
Given this is the core code.

Reply:
Anyway when 1.7.0 is gonna release, we'll have to cleanup this again. But currently this has lot of duplicate code and I only want a code here which has platform level differences.

Reply:
Can you please make a change with this?